### ISSUE TITLE
Cannot join nodes to KKP cluster with kubeadm 1.22

### ISSUE THREAD

**Issue by nightkr** [kubermatic/kubermatic#8039]
The cluster creates fine in the KKP, but trying to join a node to the cluster fails with the following error message:

```
teo@kubermatic-testcluster:~$ sudo kubeadm join 6q5tmn8kv4.kubermatic.192-168-122-184.sslip.io:32157 --token [snip] --discovery-token-ca-cert-hash sha256:[snip] 
[preflight] Running pre-flight checks
[preflight] Reading configuration from the cluster...
[preflight] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
error execution phase preflight: unable to fetch the kubeadm-config ConfigMap: failed to decode cluster configuration data: no kind "ClusterConfiguration" is registered for version "kubeadm.k8s.io/v1beta1" in scheme "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme/scheme.go:32"
To see the stack trace of this error execute with --v=5 or higher
```

Looking at the ConfigMap kube-system.kubeadm-config, we can verify that it is indeed stored as a ClusterConfiguration v1beta1:
```json
{
	"ClusterConfiguration": "apiServer:
		  extraArgs:
		    authorization-mode: Webhook
		  timeoutForControlPlane: 4m0s
		apiVersion: kubeadm.k8s.io/v1beta1
		certificatesDir: /etc/kubernetes/pki
		imageRepository: k8s.gcr.io
		kind: ClusterConfiguration
		kubernetesVersion: 1.22.2
		",
	"ClusterStatus": "apiVersion: kubeadm.k8s.io/v1beta1
		kind: ClusterStatus
		apiEndpoints:
		"
}
```

Looking at the release notes, support for kubeadm.k8s.io/v1beta1 was dropped in kubeadm 1.22 (in favour of v1beta2 and v1beta3).

Versions:
kubeadm version: &version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:37:34Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
child k8s cluster version: 1.22.2
kubermatic: v2.18.1 community edition

**Comment by embik** [comment]
/kind bug
/sig cluster-management
/assign @embik 

I'll take a look at this.

**Comment by embik** [comment]
Thanks @teozkr for reporting this, we're going to backport the fix to this to the 2.18 release as well, but since a 2.18 patch release just came out, it might take bit for the next patch release to include this fix.
